pandas/core/reshape/reshape.py

```diff
@@ -470,9 +470,8 @@ def unstack(obj: Series | DataFrame, level, fill_value=None):
         else:
             level = level[0]
 
-    # Prioritize integer interpretation (GH #21677):
     if not is_integer(level) and not level == "__placeholder__":
-        level = obj.index._get_level_number(level)
+        obj.index._get_level_number(level)
```

Comment:
We have to check if the level exists, but the actual conversion is done later

Reply:
Could you add comment with this explanation?

Also do we have tests where this call raises to the user? (Guesing `df.unstack(level="doesnt-exist")`)

Reply:
Added.

Yes, thats how I found out :)